test/14_legendre.jl

```diff
@@ -0,0 +1,257 @@
+using Polynomials
+
+@testset "legendre and related functions" begin
+    @testset "legendreP" begin
+        n_poly  = 6
+        c       = zeros(n_poly, n_poly)
+        c[1,1]  = 1
+        c[2,2]  = 1
+        c[3, 1:3] .= [-1,  0,   3               ] / 2
+        c[4, 1:4] .= [ 0, -3,   0,   5          ] / 2
+        c[5, 1:5] .= [ 3,  0, -30,   0, 35      ] / 8
+        c[6, 1:6] .= [ 0, 15,   0, -70,  0, 63  ] / 8
+
+        n_x = 20
+        x_arr = range(-1, stop=1, length=n_x)
+        for n = 0:n_poly-1
+            P = Poly(c[n+1,:])
+            for x in x_arr
+                @test legendreP(n, x) ≈ P(x)        rtol=1e-14
+            end
+        end
+
+        @test_throws DomainError legendreP(-1, 0)
+        @test_throws DomainError legendreP( 1, 2)
+        @test_throws DomainError legendreP(-1, 2)
```

Comment:
These domain errors are wrong as the functions are well-defined, we should follow Mathematica:
```mathematica
In[1]:= LegendreP[-1, 0]

Out[1]= 1

In[2]:= LegendreP[1, 2]

Out[2]= 2

In[3]:= LegendreP[-1, 2]

Out[3]= 1
```

Reply:
What do you think about finalizing the current pull request and adding larger domains (w.r.t n and x) in another pull request/issue?

Reply:
For `k < 0` (or for that matter non-integer `k`): sure, a seperate PR makes sense.

For `! (-1 ≤ x ≤ 1)`: it is simply a matter of deleting the error from being thrown. The recurrence is exactly same and is stable.

Reply:
done. see commit [d777dcc](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/d777dccb39a75d7a123e52222994945ee47231c3)